rcl_yaml_param_parser/include/rcl_yaml_param_parser/parser.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL_YAML_PARAM_PARSER__PARSER_H_
+#define RCL_YAML_PARAM_PARSER__PARSER_H_
+
+#include <stdlib.h>
+
+#include "rcl_yaml_param_parser/types.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/// \brief Parse the YAML file and populate params_hdl
+/// \param[in] file_path is the path to the YAML file
+/// \param[out] params_st points to the populated paramter struct
+/// \return true on success and false on failure
+bool parse_yaml_file(const char * file_path, rcl_params_t * params_st);
+
+/// \brief Free param structure
+/// \param[in] params_st points to the populated paramter struct
+/// \param[in] allocator memeory allocator to be used
+void free_node_struct(rcl_params_t * params_st, const rcutils_allocator_t allocator);
```

Comment:
We usually use functions named `_init` and `_fini` for structure allocation/deallocation

Reply:
Agree! Fixed it